Forge.TreeWalker/src/TreeNodeContext.cs

```diff
@@ -0,0 +1,102 @@
+//-----------------------------------------------------------------------
+// <copyright file="TreeNodeContext.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+// <summary>
+//     The TreeNodeContext class.
+// </summary>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.Forge.TreeWalker
+{
+    using System;
+    using System.Threading;
+
+    /// <summary>
+    /// The TreeNodeContext class object is passed to Callbacks.BeforeVisitNode and Callbacks.AfterVisitNode.
```

Comment:
>The TreeNodeContext class object is passed to Callbacks.BeforeVisitNode and Callbacks.AfterVisitNode. [](start = 8, length = 101)

The TreeNodeContext class holds relevant information about the tree node and session.
This object gets passed to CallbacksV2.BeforeVisitNode and CallbacksV2.AfterVisitNode.

Reply:
updated

---
In reply to: [506634857](https://github.com/microsoft/Forge/pull/32#discussion_r506634857) [](ancestors = 506634857)